src/System.Text.Encoding.CodePages/src/System/Text/SBCSCodePageEncoding.cs

```diff
@@ -867,10 +868,7 @@ public override unsafe int GetCharCount(byte* bytes, int count, DecoderNLS decod
                     // Make sure we have a fallback buffer
                     if (fallbackBuffer == null)
                     {
-                        if (decoder == null)
```

Comment:
cc: @tarekgh 
By this point we've already dereferenced decoder above when accessing its FallbackBuffer, so we know it's not null